### PR TITLE
PubNX new standalone adapter

### PR DESCRIPTION
**Pull request by prebid-pubnx**
## Type of change

- [x] New bidder adapter   

## Description of change

New Bidder Adapter

- test parameters for validating bids
```
{
  bidder: 'pubnx',
  params: {
   placementId:PNX-HB-E670793V6BHHH8
  }
}
```

- contact email of the adapter’s maintainer
- [ prebid-team@pubnx.com ] official adapter submission

### CODE REVIEW

**Comment by bretg** [comment]
Note that since `pubnx` is currently an alias for serverbidBidAdapter, there are already docs.

**Comment by mike-chowla** [comment]
I'm not getting any bids back from the test parameters.  Also, the test paras aren't formatted correctly.  I'm assuming PNX987 should be "PNX987" so it's a string

Request URL: http://hb.pubnxserv.com/vzhbidder/bid?

Request Body:
`{  
 "q":"%7B%22_vzPlacementId%22%3A%22PNX987%22%2C%22_rqsrc%22%3A%22http%3A%2F%2Fmike.pubmatic.com%3A9999%2FintegrationExamples%2Fgpt%2Fhello_world.html%3Fpbjs_debug%3Dtrue%22%2C%22_cb%22%3A1543376260%2C%22_slotBidId%22%3A%22290f0659a5efe9%22%2C%22_cbn%22%3A%22pbjs%22%7D"
}`

Response Body:
`
{"slotBidId":"290f0659a5efe9","statusText":"Placement_ID_not_found"}
`

**Comment by prebid-pubnx** [comment]
hi @mike-chowla  ,
      we have hostname validation on our platform that's why you are not getting any response, we have created new placement id for pubmatic.com. you can use "PNX-HB-E670793V6BHHH8" as placement id and you will get response.

**Comment by prebid-pubnx** [comment]
Hi,

We have hostname validation on our platform that's why you are not 
getting any response, we have created new placement id for pubmatic.com. 
you can use "PNX-HB-E670793V6BHHH8" as placement id and you will get 
response.


Regards,
Satyaprakash Mishra

**Comment by prebid-pubnx** [comment]
Hi @mike-chowla  any progress on that placement ID?

**Comment by mike-chowla** [comment]
I tried the new placement id but am still not getting bid:

Request:
`{  
   "q":"%7B%22_vzPlacementId%22%3A%22PNX-HB-E670793V6BHHH8%22%2C%22_rqsrc%22%3A%22http%3A%2F%2Fmike.pubmatic.com%3A9999%2FintegrationExamples%2Fgpt%2Fhello_world.html%3Fpbjs_debug%3Dtrue%22%2C%22_cb%22%3A1544225316%2C%22_slotBidId%22%3A%2226482703a33431%22%2C%22_cbn%22%3A%22pbjs%22%7D"
}`

Response:
`{"slotBidId":"26482703a33431","statusText":"Application_Error"}`

**Comment by prebid-pubnx** [comment]
@mike-chowla 
application error is generated because of host name contains port number, either please try to remove port number from domain or you can test it on http://filmgrapevine.com/prebid-test/

**Comment by prebid-pubnx** [comment]
Hi @mike-chowla ,

We tried at out end with above request shared by you and we could see bid received and ad rendered at our end. Can you just check again hitting the same url for some time. And if issue still persists then can you please share request and response both?

**Comment by prebid-pubnx** [comment]
@mike-chowla Are you able to receive any bids now? Please share request and response both if you are getting the same.

**Comment by prebid-pubnx** [comment]
@mike-chowla Please check now.

**Comment by prebid-pubnx** [comment]
@mike-chowla  any success??

**Comment by prebid-pubnx** [comment]
@mike-chowla we could see on our end that request contains port number. As requested earlier can you again check without port number please?

**Comment by mike-chowla** [comment]
Still doesn't work

Request
`{"q":"%7B%22_vzPlacementId%22%3A%22PNX-HB-E670793V6BHHH8%22%2C%22_rqsrc%22%3A%22http%3A%2F%2Fmike.pubmatic.com%2FintegrationExamples%2Fgpt%2Fhello_world.html%22%2C%22_cb%22%3A1546998236%2C%22_slotBidId%22%3A%222961cf6389e2ac%22%2C%22_cbn%22%3A%22pbjs%22%7D"}`

Response:
`{"vzhPlacementId":"PNX-HB-E670793V6BHHH8","slotBidId":"2d952eeef02508","statusText":"NO_BIDS"}`

File: modules/pubnxBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      }
+
+      let reqSrc = utils.getTopWindowLocation().href;
+      var pnxReq = {
+        _vzPlacementId: placementId,
+        _rqsrc: reqSrc,
+        _cb: cb,
+        _slotBidId: slotBidId,
+        _cpm: cpm,
+        _cbn: '$$PREBID_GLOBAL$$'
```
Comment (mike-chowla) [inline modules/pubnxBidAdapter.js]: Adapters may not use the $$PREBID_GLOBAL$$ variable: Instead, they must load any necessary functions and call them directly.
http://prebid.org/dev-docs/bidder-adaptor.html

It's not at all clear to me why are you referencing $$PREBID_GLOBAL$$ here

**Comment by mike-chowla** [comment]
I was able to load to test add.  As I was doing my final review of the code, I noticed you were referencing $$PREBID_GLOBAL$$ which is not permitted

**Comment by prebid-pubnx** [comment]
We are unable to find our adapter(PubNX) in prebid.js release 1.38.0 when we select PubNX and download prebid.js . Can you check with it.

**Comment by prebid-pubnx** [comment]
Hi Bret,

Our new PubNX adapter code has been merged a week ago and yesterday a 
new version (1.38.0) of prebid.js was released.
We could see our adapter listed in download page. But when we download 
prebid.js with pubnx selected, we could not see our adapter in that. We 
have put a comment on ticket #3298 also.

As this is an urgency for us, can someone please check this one.

Just FYI, Earlier PubNX was released under ServerBid.

Thanks & Regards,

Satyaprakash Mishra
Team PubNX

**Comment by bretg** [comment]
@prebid-pubnx -- don't know why the Prebid.org download page isn't working for this adapter. @mkendall07 will need to look into it.

However, the workaround is straightforward enoough: just build from the command line. See https://github.com/prebid/Prebid.js/blob/master/README.md and run `gulp build --modules pubnxBidAdapter,currency` or whatever list of modules you need. The result in build/dist/prebid.js is what you want.

**Comment by prebid-pubnx** [comment]
Thanks @bretg Bret for the information,
thanks @amuraco

**Comment by prebid-pubnx** [comment]
Hi @bretg , when can we expect this thing to be resolved.

**Pull request merged**